lib/basedriver/commands/event.js

```diff
@@ -14,13 +16,25 @@ commands.logCustomEvent = function (vendor, event) {
 /**
  * Get the event log
  *
- * @param {string} type - currently unused but a placeholder for a future
- * feature where filtering of log events is allowed
+ * @param {?string|Array<string>} type - the event type to filter with.
+ * It returns all events if the type is not provided or empty string/array.
  * @returns {object} - the event history log object
  */
-commands.getLogEvents = function (/*type*/) {
-  // type is currently unused but may be implemented for filtering later on
-  return this._eventHistory;
+commands.getLogEvents = function (type = null) {
+  if (_.isEmpty(type)) {
+    return this._eventHistory;
+  }
+
+  if (!_.isArray(type)) {
+    type = [type];
+  }
+
+  return _.reduce(this._eventHistory, function a (acc, eventTimes, eventType) {
```

Comment:
why not to use an anonymous function here? `(acc, eventTimes, eventType) => { .. }`

Reply:
oops, that was my wrong commit when i checked something..